python/arcticdb/version_store/helper.py

```diff
@@ -272,10 +292,20 @@ def get_s3_proto(
         s3.ca_cert_dir = ca_cert_dir
     if ssl is not None:
         s3.ssl = ssl
+    
+    if aws_auth == AWSAuthMethod.STS_PROFILE_CREDENTIALS_PROVIDER or aws_profile:
+        if is_nfs_layout:
+            raise ArcticNativeException("aws_auth and aws_profile can only be set for S3")
```

Comment:
These should be user input errors https://docs.arcticdb.io/latest/error_messages/#user-input-errors